Oleksandr.Svidovyi/Task_5/src/main/java/com/hillel/task_5/servlet/ClientServlet.java

```diff
@@ -0,0 +1,165 @@
+                Integer clientId = currentId;
+                Client client = new Client(clientId, requestModel.getClientName());
+                clientRepository.getClients().put(clientId, client);
+                currentId++;
+
+                resp.getWriter().println("CLient was added \n" + gson.toJson(clientRepository.getClients().get(clientId)));
+            } else {
+                resp.getWriter().println("Please, enter client's name");
+                resp.setStatus(400);
+            }
+        } catch (Exception e) {
+            log.error("Exception while trying creating a client");
+            resp.getWriter().println("Something went wrong. Please, try again.");
+            resp.setStatus(400);
+        }
+
+    }
+
+
+    @Override
+    protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json");
+        Gson gson = new Gson();
+        clientRepository = ClientRepository.getInstance();
+
+        try {
+            ClientPutRequestModel requestModel = gson.fromJson(req.getReader(), ClientPutRequestModel.class);
+
+            if (requestModel.getClientId() != null && !requestModel.getClientId().equals("")
+                    && requestModel.getClientName() != null && !requestModel.getClientName().equals("")) {
+                Integer clientId = Integer.decode(requestModel.getClientId());
+
+                if (clientId <= clientRepository.getClients().size()) {
+                    Client client = clientRepository.getClients().get(clientId);
+                    client.setName(requestModel.getClientName());
+
+                    if (client != null) {
+                        resp.getWriter().println("CLient was edited \n" + gson.toJson(clientRepository.getClients().get(clientId)));
+                    } else {
+                        resp.getWriter().println("Client not found.");
+                        resp.setStatus(404);
+                    }
+                } else {
+                    resp.getWriter().println("The client's id is not valid");
+                    resp.setStatus(400);
+                }
+            } else {
+                resp.getWriter().println("Please, enter client's name");
+                resp.setStatus(400);
+            }
+        } catch (Exception e) {
+            log.error("Exception while trying creating a client");
+            resp.getWriter().println("Something went wrong. Please, try again.");
+            resp.setStatus(400);
+        }
+    }
+
+
+    @Override
+    protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json");
+        Gson gson = new Gson();
+        clientRepository = ClientRepository.getInstance();
+
+        try {
+            ClientGetRequestModel requestModel = gson.fromJson(req.getReader(), ClientGetRequestModel.class);
```

Comment:
Use req.getParameter("clientId")

Reply:
We can't map delete method from HTML therefore I use DeleteClientRequestModel to fetch data from Postman.

Reply:
Why HTML? This task is not for UI, but for methods, which can be called via Postman